Задача 1 

In [47]:
a = [100, 80, 75, 77, 89, 33, 45, 25, 65, 17, 30, 230, 24, 57, 55, 70, 75, 65, 84, 90, 150]
a = sorted(a)
average_ = sum(a)/len(a)

list_ = [num**2 - average_**2 for num in a]

(sum(list_)/(len(a)-1))**0.5, average_

(47.3479521355314, 73.14285714285714)

(47.3479521355314, 73.14285714285714)

Ответ: Среднее квадратичное отклонение 47.3 

Задача 2.

Ищем точки квартилей. По скольку длина списка нечётная, то квартили совпадают с точками массива. 
Делим длину списка последовательно на 2, округляем вниз и делим ещё раз.
По скольку итерация индексов списка идёт с нуля, как раз получаем нужные индексы квартилей в листе. 

In [48]:
q1, q3 = a[5], a[-6]
q1, q3 

(45, 84)

In [49]:
iqr = q3 - q1
bottom = q1 - 1.5*iqr
top = q3 + 1.5*iqr
bottom, top

(-13.5, 142.5)

(-13.5, 142.5) 
По скольку отрицательных значений в перечне нет, а список уже отсортирован по возрастающей, проверяем на условие больше 142,5 и забираем индексы совпадений. 

In [50]:
counter_ = len([1 for i in a if i> top])
counter_

2

Ответ: два значения выбиваются из заданной границы сверху. 


Задача 3 

Тут не очень уверен, теория вероятностей такая запутанная когда что применять.
В голову приходит только, что у нас одновременно происходит два события:
    1) мы выбрали студента из одной из трёх групп
    2) студент сдал экзамен
    
Поэтому мы будем перемножать вероятности событий.
А: 0.6*100/235 = 0,25
В: 0,7*90/235 = 0,268
С: 0,8*45/235 = 0,153

Наибольшая вероятность что студент из группы В. 

Ответ: 2 В